src/components/modules/renderer.ts

```diff
@@ -46,6 +46,15 @@ export default class Renderer extends Module {
    * @param {OutputBlockData[]} blocks - blocks to render
    */
   public async render(blocks: OutputBlockData[]): Promise<void> {
+
+    if (blocks.length === 0) {
+      blocks = [{
+          "data": {"text": ""},
+          "id": "WhateverXX",
```

Comment:
Consider leaving a comment about why this initialization is required, and the significance of the initial values if any (eg. "Placeholder values will not appear on screen" or "text field of data must be given this value for this reason").